lib/master_manipulator/service.rb

```diff
@@ -2,71 +2,227 @@
 
 module MasterManipulator
   module Service
-
     # Restart the puppet server and wait for it to come back up
     # ==== Attributes
     # *+host+ - the host that this should operate on
     # *+opts+ - an options hash - not required
     #   *+:wait_cycle+ - the number of cycles to attempt retry
     #   *+:is_pe?+ - Boolean : if the SUT is PE, defaults to true
+    #   *+:hup?+ - Boolean : if true, reload server; if false, restart server; defaults to false
     #
     # Raises a standard error if the wait is unsuccessful
     #
     # ==== Example
     # restart_puppet_server(master)
     # restart_puppet_server(master, {:wait_cycles => 20})
-    def restart_puppet_server(host, opts = {})
-
+    # restart_puppet_server(master, {:is_pe? => false, :hup? => true})
+    def restart_puppet_server(host, opts = {:is_pe? => true, :hup? => false})
       start_time = Time.now
 
-      opts[:is_pe?] ||= true
       opts[:is_pe?] ? service_name = 'pe-puppetserver' : service_name = 'puppetserver'
-
-      on(host, puppet("resource service #{service_name} ensure=stopped"))
-      on(host, puppet("resource service #{service_name} ensure=running"))
-      hostname = on(host, 'hostname').stdout.chomp
-      opts[:wait_cycles] ||= 10
-
-      pe_ver = pe_version(host)
-      three_eight_regex = /^3\.8/
-
-      # This logic is not ideal refactor in the future
-      # if its PE and not 3.8 use the status endpoint
-      # it its not PE or it is 3.8 use the simple curl call
-      if pe_ver && !pe_ver.match(three_eight_regex)
-        curl_call = "-k -X GET -H 'Content-Type: application/json' https://#{hostname}:8140/status/v1/services?level=debug"
+    
+      if opts[:hup?]
+        hup_server(host, service_name)
+        return
       else
-        curl_call = "-I -k https://#{hostname}:8140"
+        on(host, puppet("resource service #{service_name} ensure=stopped"))
+        on(host, puppet("resource service #{service_name} ensure=running"))
       end
 
-      (1..opts[:wait_cycles]).each do |i|
-
-        @result = curl_on(host, curl_call, :acceptable_exit_codes => [0,1,7])
-        # parse body if we are using PE and we are not in PE 3.8
-        (pe_ver && !pe_ver.match(three_eight_regex)) ? @body = JSON.parse(@result.stdout) : @body = []
+      # FIXME: This logic will need re-work after 3.8 goes EOL at the end
+      # of 2016 to remove the 3.8 references. That will simplify the code
+      # path, too.
+      if opts[:is_pe?]
+        hostname = on(host, 'hostname').stdout.strip
```

Comment:
I think we should get the hostname out of the options hash rather than making another ssh call to the host and getting the hostname.  I think the beaker's previous problems with the hostname have been resolved.